### PR TITLE
Fix Exporting RNN/LSTM's Initial State (h0/c0) to ONNX

### PR DESCRIPTION
**Pull request by lara-hdr**

### CODE REVIEW

**Comment by jerryzh168** [comment]
please fix merge conflict

**Review by houseroad — COMMENTED** [review]
is this PR based on PR #22993?

**Comment by houseroad** [comment]
Btw, let's rebase again.

**Comment by lara-hdr** [comment]
@bddppq could you please take a look when you have the time? 
I'll resolve the conflicts today.

**Comment by houseroad** [comment]
friendly ping? @lara-hdr

**Comment by lara-hdr** [comment]
> friendly ping? @lara-hdr

Haven't had a chance to come back to this one. Will rebase and merge conflicts tonight.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -61,14 +63,14 @@ def run_model_test(self, model, batch_size=2, state_dict=None,
...
         f = io.BytesIO()
-        torch.onnx.export(model, input, f,
-                          opset_version=self.opset_version,
-                          example_outputs=output,
-                          do_constant_folding=do_constant_folding,
-                          keep_initializers_as_inputs=self.keep_initializers_as_inputs,
-                          dynamic_axes=dynamic_axes,
-                          input_names=input_names, output_names=output_names)
-
+        torch.onnx._export(model, input, f,
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: so we didn't expose  `fixed_batch_size` to torch.onnx.export?

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 2]
```diff
@@ -426,7 +431,13 @@ void fixDefaultRNNState(Graph* graph, Node* n, int input_index, int opset_versio
...
   concated_dims->addInput(hidden_size->outputs()[0]);
 
-  if (opset_version < 9) {
+  Node* fixed_init_state = graph->create(onnx::Expand, 1);
+  fixed_init_state->insertBefore(n);
+  fixed_init_state->addInput(initial_state);
+  fixed_init_state->addInput(concated_dims->outputs()[0]);
+  n->replaceInput(input_index, fixed_init_state->outputs()[0]);
+
+  /*if (opset_version < 9) {
```
Comment (houseroad) [inline torch/csrc/jit/passes/onnx/peephole.cpp]: delete the code?

**Review by houseroad — COMMENTED** [review]
Overall looks good. Some comments to address.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (lara-hdr) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: I did not, but if you think we should I could add it.

Comment (houseroad) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Let's do it in another PR then

**Comment by lara-hdr** [comment]
@houseroad is this ready for merge?

**Pull request closed without merging**